edenia/antelope-tools
# Feat/nodes stats 1160

### Add statistics from endpoint health checks

### What does this PR do?

- Resolve #1160 
- Avoid to request the health check for the same url when several nodes share the endpoints
- Tracking the number of health checks performed per day and how many times the endpoints have responded.
- Save the response time from Costa Rica and show the top 5 fastest endpoints.
- New page with statistics of endpoints per producer. 

### Steps to test

1. Run the project locally
2. Go the new page API Endpoints Stats /endpoints-stats
3. Check the content
4. Search a producer with the Select component
5. Check that the search show the endpoints data
6. Go to endpoints list page
7. Click the icon next to the producer's name
8. Check that link to the new page

#### Screenshot

## Full page

![imagen](https://user-images.githubusercontent.com/66583677/227279544-ada5cdd3-9c19-431b-8f30-2e267a026550.png)

## Fastest endpoints

![imagen](https://user-images.githubusercontent.com/66583677/227279565-e887cec8-5d6a-4a76-bac1-8a8a34d6ccf2.png)

## List of endpoints

![imagen](https://user-images.githubusercontent.com/66583677/227279598-7b78d557-8308-44ab-be07-058f7a13bb92.png)

## Response time graph from costa rica

![imagen](https://user-images.githubusercontent.com/66583677/227279651-013fe4eb-d8b3-4f60-b580-3f3452b5f0c7.png)
![imagen](https://user-images.githubusercontent.com/66583677/227298182-738c8e5c-6241-4f2f-85d5-9f45807ab69e.png)


#### CheckList
